### PR TITLE
fix(module: tree): fix default keys bug with setTimeOut

### LINKED ISSUE #3001: tree component nzSelectedKeys not work with setTimeout
### Reproduction link
[https://stackblitz.com/edit/angular-mg6hvr-cesjkx?file=src/app/app.component.ts](https://stackblitz.com/edit/angular-mg6hvr-cesjkx?file=src/app/app.component.ts)

### Steps to reproduce
like demo

### What is expected?
nzSelectedKeys(include nzCheckedKeys, nzExpandedKeys) render tree dom correctly

### What is actually happening?
read demo

| Environment | Info |
|---|---|
| ng-zorro-antd | 7.0.0 |
| Browser | any |

### PR DESCRIPTION
**Pull request by simplejason**
close #3001

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Application (the showcase website) / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: #3001 

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: components/tree/nz-tree-base.service.ts [hunk 1]
```diff
@@ -193,6 +196,7 @@ export class NzTreeBaseService implements OnDestroy {
         this.selectedNodeList = [ node ];
       }
     }
+    index = this.selectedNodeList.findIndex(n => node.key === n.key);
```
Comment (HsuanXyz) [inline components/tree/nz-tree-base.service.ts]: 这句代码是必须的吗？

File: components/tree/nz-tree-node.component.ts [hunk 2]
```diff
@@ -392,7 +366,7 @@ export class NzTreeNodeComponent implements OnInit, OnChanges, OnDestroy {
     e.stopPropagation();
     this.ngZone.run(() => {
       this.clearDragClass();
-      if (this.nzTreeService.getSelectedNode() === this.nzTreeNode) {
+      if (this.nzTreeService.getSelectedNode().key === this.nzTreeNode.key) {
```
Comment (HsuanXyz) [inline components/tree/nz-tree-node.component.ts]: `this.nzTreeService.getSelectedNode()` 不是一定有值的，如果向 tree-node 拖拽其他内容（比如图片或文字）就会报错

File: components/tree/nz-tree-node.ts [hunk 3]
```diff
@@ -19,127 +19,265 @@ export interface NzTreeNodeOptions {
...
+  set title(value: string) {
+    this._title = value;
+    this.update();
+  }
+
+  get key(): string {
+    return this._key;
+  }
+
+  set key(value: string) {
```
Comment (HsuanXyz) [inline components/tree/nz-tree-node.ts]: 有些 getter setter 没有额外操作，可否去掉？

File: components/tree/nz-tree.component.ts [hunk 4]
```diff
@@ -65,17 +66,27 @@ export class NzTreeComponent implements OnInit, OnChanges, OnDestroy, ControlVal
...
    * @deprecated use
    * nzExpandAll instead
    */
-  @Input() @InputBoolean() nzDefaultExpandAll: boolean = false;
+  @Input() @InputBoolean() nzDefaultExpandAll = false;
   @Input() nzBeforeDrop: (confirm: NzFormatBeforeDropEvent) => Observable<boolean>;
 
+  @Input() @InputBoolean()
+  set nzMultiple(value: boolean) {
+    console.log(value);
```
Comment (HsuanXyz) [inline components/tree/nz-tree.component.ts]: `console ` 去掉

File: components/tree-select/nz-tree-select.component.ts [hunk 5]
```diff
@@ -274,7 +274,7 @@ export class NzTreeSelectComponent implements ControlValueAccessor, OnInit, OnDe
     node.isChecked = false;
     if (this.nzCheckable) {
       this.nzTreeService.conduct(node);
-      this.nzTreeService.setCheckedNodeList(node);
+      // this.nzTreeService.setCheckedNodeList(node);
```
Comment (HsuanXyz) [inline components/tree-select/nz-tree-select.component.ts]: 注释可以移除掉

File: components/tree-select/nz-tree-select.component.ts [hunk 6]
```diff
@@ -355,6 +356,8 @@ export class NzTreeSelectComponent implements ControlValueAccessor, OnInit, OnDe
       }
     }
     this.selectedNodes = [ ...(this.nzCheckable ? this.nzTreeService.getCheckedNodeList() : this.nzTreeService.getSelectedNodeList()) ];
+    // console.log(this.nzCheckable, this.selectedNodes.length, this.selectedNodes, this.value, this.value.length, '===');
```
Comment (HsuanXyz) [inline components/tree-select/nz-tree-select.component.ts]: 注释可以移除掉

File: components/tree/nz-tree-base.service.ts (hunk 1 above)
Comment (simplejason) [reply components/tree/nz-tree-base.service.ts]: 上面的逻辑可能导致数组改变，重新查询

**Pull request merged**